Realm/RLMRealm.mm

```diff
@@ -276,12 +289,15 @@ + (void)asyncOpenWithConfiguration:(RLMRealmConfiguration *)configuration
         }
     };
 
-    dispatch_async(queue, ^{
+    RLMAsyncOpenTask *ret = [RLMAsyncOpenTask new];
+    dispatch_async(s_async_open_queue, ^{
```

Comment:
Why move this to its own queue?

Reply:
The only change here is to let the tests override the queue used, so that they can make it use a serial queue rather than the default of a concurrent queue to ensure that things happen in the desired order.

If the question is just "why is there a background queue involved at all?" then it's because there is a bit of i/o and cpu done on the source thread even as part of get_synchronized_realm().